mastodon/accounts.py

```diff
@@ -427,10 +427,10 @@ def account_update_credentials(self, display_name=None, note=None,
                 raise MastodonIllegalArgumentError(
                     'A maximum of four fields are allowed.')
 
-            fields_attributes = []
-            for idx, (field_name, field_value) in enumerate(fields):
-                params_initial[f'fields_attributes[{idx}][name]'] = field_name
-                params_initial[f'fields_attributes[{idx}][value]'] = field_value
+            for idx, (field_name, field_value, field_verified_at) in enumerate(fields):
+                params_initial[f'fields_attributes[{idx}][name]'] = fields[idx][field_name]

```

Comment:
Swooping in from your mastodon post here: `fields[idx]` should always be a tuple because fields is a list of tuples. So you're trying to grad `[field_name]` as an indice of a tuple which you can't do.

You're iterating over fields, why not just `params_initial[f'fields_attributes[{idx}][name]'] =  field_name`?

Reply:
@astrowonk Thank you! I have a better understanding of how the code works now. I updated my PR and also the tests to account for the new `verified_at` attribute.